### DISCUSSION TITLE
Installation Problem

### DISCUSSION THREAD

**Discussion by serhatyaris34** [emgucv/emgucv#636] [Q&A] [answered]
Hello all, I am very new with Emgu and trying to follow tutorials. My problem is about finding some necessary files such as Emgu.CV.dll, Emgu.CV.UI.dll. I followed this official quide(https://www.emgu.com/wiki/index.php/Download_And_Installation) in order to start using Emgu. In the quide, I attempted this section =>(1.2Using the Downloadable packages). By helping this quide, I reached to this website(https://github.com/emgucv/emgucv/releases) to download necessary files. But when I downloaded zip files and unzip them. I can't see any 'bin'  files as my tutorials show. Ofcourse other necassary files, too. Since I can't access those files; I can't go forward with the library. Where I might make mistake? (Besides that same issue occurs when I use libemgucv-windesktop-4.5.4.4788.exe instead of zip/unzip technique) Again I can't see any bin files?

**Comment by emgucv** [comment]
The binary files are located under the "libs" folder in the ".zip" packages. If installed using the ".exe" file, they are located under the "{install_folder}\libs" folder.

It may be easier to use the nuget packages instead of the binaries from the zip packages.

**Reply by serhatyaris34** [reply]
Thank you for quick reply. However, I have still 2 questions.

Question 1:
First one is related to "unzip of libemgucv-windesktop-4.5.4.4788.zip". When I unzip the file; there is no "libs" folder in it. I share of a picture where you can see the folders after I unzip. After all, isn't those files(Emgu.CV.dll, Emgu.CV.UI.dll) supposed to be somewhere there? When I try to "Find in the Folder" inside of all unziped folders for those files(Emgu.CV.dll, Emgu.CV.UI.dll), I don't see those files at all.

![GitHub_Emgu](https://user-images.githubusercontent.com/32990026/140026901-b32d8cc1-29e6-4c35-8a4a-dea444ed15c6.png)
 
Question 2: When I prefered nuget packages instead of using binaries, I downloaded "Emgu.CV~4.5.4.4788" from NuGet tool. But I can't access "ToBitmap()" function. What might be problem?
```
            using Emgu.CV;
            using Emgu.CV.Structure;
             ***********
            Image<Bgr, byte> imf = new Image<Bgr, byte>(fileName);
            pictureBox1.Image = imf.ToBitmap(); // IT DOES NOT RECOGNIZE the ToBitmap func!
```
EDIT QUESTION 2: I saw one of the commands in stackoverflow(https://stackoverflow.com/questions/15406235/convert-emgucv-image-to-system-drawing-bitmap/15413356) goes like that: ToBitmap functions are moved another nuget called "Emgu.CV.Bitmap". So I downloaded this nuget too besides "Emgu.CV". And my IDE can recognize the ToBitmap function now. However when I run my program, I get this error at this time:
![github_emgu_error](https://user-images.githubusercontent.com/32990026/140078910-e23b8fff-2d27-48f9-92d3-77472434ea11.png)

(I have already set build option to x64 btw) Why I get this error? Thanks in advance!

**Comment by emgucv** [comment, marked as answer]
RE: Question 1:
Did you download the file from here?
https://github.com/emgucv/emgucv/releases/tag/4.5.4

The direct link to the zip file is:
https://github.com/emgucv/emgucv/releases/download/4.5.4/libemgucv-windesktop-4.5.4.4788.zip

Once it is downloaded and unzipped, the folder looks like this:
![image](https://user-images.githubusercontent.com/2035816/140424944-ab7cd704-eb6a-4bf0-8be5-75bc716cca8e.png)

So the "libs" folder is definitely there.

RE: Questions 2:
Install the Emgu.CV.Bitmap nuget packages if you want to convert Mat to/from Bitmap.
https://www.nuget.org/packages/Emgu.CV.Bitmap/ 

Instructions for using the nuget package and trouble shooting the PInvoke errors can be found here:
https://www.emgu.com/wiki/index.php/Download_And_Installation
